pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -648,11 +648,21 @@ func managePod(ctx context.Context, configMapsGetter corev1client.ConfigMapsGett
 		return nil, false, fmt.Errorf("couldn't get the proxy config from observedConfig: %v", err)
 	}
 
-	proxyEnvVars := proxyMapToEnvVars(proxyConfig)
+	proxyEnvVars := stringMapToEnvVars(proxyConfig)
 	for i, container := range required.Spec.Containers {
 		required.Spec.Containers[i].Env = append(container.Env, proxyEnvVars...)
 	}
 
+	// Set OPENSHIFT_DO_VSPHERE_MIGRATION to 'true' if vSphere migration is enabled
+	storageConfig, _, err := unstructured.NestedStringMap(observedConfig, "targetconfigcontroller", "storage")
+	if err != nil {
+		return nil, false, fmt.Errorf("couldn't get the storage config from observedConfig: %v", err)
+	}
+	storageEnvVars := stringMapToEnvVars(storageConfig)
+	for i, container := range required.Spec.Containers {
+		required.Spec.Containers[i].Env = append(container.Env, storageEnvVars...)
```

Comment:
we need this only on the KCM container (index 0): https://github.com/openshift/kubernetes/pull/1514

Reply:
We are using the index approach for other manipulations as well, so it should be fine. I think the main benefit is the pod yaml will be less cluttery / easier to read.